plugin/trino-hive/src/test/java/io/trino/plugin/hive/fs/TestTransactionScopeCachingDirectoryLister.java

```diff
@@ -159,8 +161,8 @@ private static class CountingDirectoryLister
             implements DirectoryLister
     {
         private final Map<Location, List<TrinoFileStatus>> fileStatuses;
-        private int listCount;
-        private boolean throwException;
+        private final AtomicInteger listCount = new AtomicInteger();
+        private volatile boolean throwException;
```

Comment:
nit: This can be as a prep commit right ? Before fixing the actual issue

Reply:
yep split it into 2